cpp/purify/yaml-parser.cc

```diff
@@ -28,10 +30,44 @@ YamlParser::YamlParser (const std::string& filepath)
 
 void YamlParser::readFile ()
 {
-  YAML::Node config = YAML::LoadFile(this->filepath_);
-  // A bit of defensive programming
-  assert(config.Type() == YAML::NodeType::Map);
-  this->config_file = config;
+  try
+    {
+      YAML::Node config = YAML::LoadFile(this->filepath_);
+      // A bit of defensive programming
+      assert(config.Type() == YAML::NodeType::Map);
+      this->config_file = config;
+    }
+  catch(YAML::BadFile& exception)
+    {
+      char cwd[1024];
+      getcwd(cwd, sizeof(cwd));
+      std::cerr << "The input file path " << this->filepath_ << " could not be found from " << cwd << std::endl;
+      throw (std::runtime_error("Runtime error while trying to find config.yaml"));
+    }
+}
+
+template <typename T>
+T get(const YAML::Node& node_map, const std::initializer_list<const char*> indicies)
+{
+  YAML::Node node = YAML::Clone (node_map);
+  std::string faulty_variable;
+  for (const char* index : indicies)
+    {
+      faulty_variable = std::string(index);	  
+      node = node[index];
+      if (!node.IsDefined())
+	{
+	  throw std::runtime_error("The initialisation of " + faulty_variable + " is wrong.");
+	}
+    }
+  try
+    {
```

Comment:
Check if node.size() > 1.

Reply:
If `node.size > 1`, then iterate through the container and push back the elements into the container `T<typename T:: value_type>`.

Reply:
done